gm_dotnet_managed/GmodNET/GameConsoleWriter.cs

```diff
@@ -13,9 +14,53 @@ internal class GameConsoleWriter : TextWriter
         public override string NewLine => "\n";
         public override Encoding Encoding => Encoding.UTF8;
 
-#pragma warning disable CA2101
+        public GameConsoleWriter(ILua lua)
+        {
+            if (OperatingSystem.IsWindows())
+            {
+                Msg = Msg_Tier0;
+            }
+            else if (OperatingSystem.IsLinux())
+            {
+                lua.PushSpecial(SPECIAL_TABLES.SPECIAL_GLOB);
+                lua.GetField(-1, "game");
+                lua.GetField(-1, "IsDedicated");
+                lua.MCall(0, 1);
+                bool is_dedicated = lua.GetBool(-1);
+                lua.Pop(1);
```

Comment:
Are you sure that there is only one item to pop from the stack? It seems like there are three objects: global table, game table, and bool value returned by IsDedicated. Note that `GetBool` does not pop values from the stack: [https://docs.gmodnet.xyz/api/GmodNET.API.ILua.html#GmodNET_API_ILua_GetBool_System_Int32_](https://docs.gmodnet.xyz/api/GmodNET.API.ILua.html#GmodNET_API_ILua_GetBool_System_Int32_)